pkg/trustedresources/verify.go

```diff
@@ -223,79 +197,34 @@ func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.
 	return VerificationResult{VerificationResultType: VerificationPass}
 }
 
-// doesAnyVerifierPass loop over verifiers to verify the resource, return true if any verifier pass verification.
-func doesAnyVerifierPass(resource metav1.Object, signature []byte, verifiers []signature.Verifier) bool {
+// doesAnyVerifierPass loop over verifiers to verify the checksum and the signature, return true if any verifier pass verification.
+func doesAnyVerifierPass(checksum []byte, signature []byte, verifiers []signature.Verifier) bool {
 	passVerification := false
 	for _, verifier := range verifiers {
-		// if one of the verifier passes verification, then this policy passes verification
-		if err := verifyInterface(resource, verifier, signature); err == nil {
+		if err := verifier.VerifySignature(bytes.NewReader(signature), bytes.NewReader(checksum)); err != nil {
+			// FixMe: changing %v to %w breaks integration tests.
+			fmt.Errorf("%w:%v", ErrResourceVerificationFailed, err.Error())
+		} else {
+			// if one of the verifier passes verification, then this policy passes verification
```

Comment:
I think the logic here is that if we have 1 passing verifier, we will set `passVerification = true`, so maybe we can just ignore the errors to keep it the same as before . (Or consider only logging warning).

Reply:
Thanks! Fixed. PTAL again 🙏